crates/cheatnet/tests/cheatcodes/spoof.rs

```diff
@@ -645,3 +646,393 @@ fn spoof_library_call() {
 
     assert_success!(output, vec![Felt252::from(123)]);
 }
+
+#[test]
+fn spoof_all_simple() {
+    let mut cached_state = create_cached_state();
+    let (mut blockifier_state, mut cheatnet_state) = create_cheatnet_state(&mut cached_state);
+
+    let contract_address = deploy_contract(
+        &mut blockifier_state,
+        &mut cheatnet_state,
+        "SpoofChecker",
+        &[],
+    );
+
+    let (
+        nonce_before,
+        account_address_before,
+        version_before,
+        chain_id_before,
+        max_fee_before,
+        signature_before,
+        _,
+    ) = call_mock_checker_getters(
+        &mut blockifier_state,
+        &mut cheatnet_state,
+        &contract_address,
+    );
+
+    let version = None;
+    let account_contract_address = None;
+    let max_fee = None;
+    let signature = None;
+    let transaction_hash = Some(Felt252::from(123));
+    let chain_id = None;
+    let nonce = None;
```

Comment:
I feel like it'd be good to have one of the tests mock all of the fields, but I guess it's fine

Reply:
we do check that in an other test

Reply:
we could probably still check the global though.

Reply:
added a case